lib/collection/src/shard/shard_holder.rs

```diff
@@ -19,18 +20,27 @@ use crate::shard::{ChannelService, CollectionId, PeerId, Shard, ShardId, ShardTr
 
 pub struct ShardHolder {
     shards: HashMap<ShardId, Shard>,
-    shard_transfers: HashMap<ShardId, ShardTransfer>,
+    shard_transfers: SaveOnDisk<HashMap<ShardId, ShardTransfer>>,
     temporary_shards: HashMap<ShardId, Shard>,
     ring: HashRing<ShardId>,
 }
 
 pub struct LockedShardHolder(pub RwLock<ShardHolder>);
 
 impl ShardHolder {
-    pub fn new(hashring: HashRing<ShardId>) -> Self {
+    pub fn new(collection_path: &Path, hashring: HashRing<ShardId>) -> Self {
+        let shard_transfers = match SaveOnDisk::load_or_init(
+            collection_path.join("shard_transfers"),
+        ) {
+            Ok(shard_transfers) => shard_transfers,
+            Err(err) => {
+                log::error!("Failed to load or init shard_transfers file: {err}. Using default empty map of transfers.");
+                Default::default()
```

Comment:
Why not to propagate the error? We do init empty hashmap if there is no file, right?

Reply:
Yes, if there is no file - it will be an empty hashmap. Mostly we use `expect` in collection load to panic if it can't load some parts, though as we discussed with you we should minimize the number of panics. I am open to suggestions as to how to handle this case.

Reply:
this part is also used in the creation of the collection, which is part of the regular serving process

Reply:
In that case actually it cannot return any error, as it will simply call `default` inside of the function. I can add a comment here about it.